src/transport.rs

```diff
@@ -71,15 +63,14 @@ impl Transport {
         method: Method,
         endpoint: &str,
         body: Option<(B, Mime)>,
-    ) -> Result<String>
+    ) -> impl Future<Item = String, Error = Error>
     where
         B: Into<Body>,
     {
-        let mut res = self.stream(method, endpoint, body)?;
-        let mut body = String::new();
-        res.read_to_string(&mut body)?;
-        debug!("{} raw response: {}", endpoint, body);
-        Ok(body)
+        self.stream(method, endpoint, body)
+            .concat2()
+            .and_then(|v| String::from_utf8(v).map_err(Error::Encoding).into_future())
+        // debug!("{} raw response: {}", endpoint, body);
```

Comment:
What do you think about keeping the `debug!` log? I've often found these useful in cases where I need to reproduce a deserialization problem with a type mapping. The errors that bubble up from serde sometimes do not contain enough contextual information to give you an idea of what is in error.